source/common/http/conn_manager_impl.cc

```diff
@@ -1048,8 +1078,18 @@ void ConnectionManagerImpl::ActiveStreamFilterBase::clearRouteCache() {
   parent_.cached_route_ = Optional<Router::RouteConstSharedPtr>();
 }
 
-void ConnectionManagerImpl::ActiveStreamDecoderFilter::addDecodedData(Buffer::Instance& data) {
-  parent_.addDecodedData(*this, data);
+Buffer::WatermarkBufferPtr ConnectionManagerImpl::ActiveStreamDecoderFilter::createBuffer() {
+  auto buffer = Buffer::WatermarkBufferPtr{
+      new Buffer::WatermarkBuffer(Buffer::InstancePtr{new Buffer::OwnedImpl()},
```

Comment:
Aside comment unrelated to this review: It would be nice if we could in general make WatermarkBuffers not require 2 allocations. Can we make WatermarkBuffer derive from OwnedImpl and potentially just override in tests via a TestWatermarkBuffer or something like that? If you don't mind can you open up an issue on this and tag with perf so we can track it for later?